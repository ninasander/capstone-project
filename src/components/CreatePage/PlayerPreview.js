import React from 'react'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import DisplayPlayerAndEnemy from '../EncounterPage/DisplayPlayerAndEnemy'

PlayerPreview.propTypes = {
  creatureEntry: PropTypes.object.isRequired,
  handleDelete: PropTypes.func.isRequired,
  handleEdit: PropTypes.func.isRequired,
}

export default function PlayerPreview({
  handleDelete,
  handleEdit,
  creatureEntry,
}) {
  return (
    <>
      <PlayerEntryStyled>
        <DisplayPlayerAndEnemy
          initiative={creatureEntry.playerInitiative}
          name={creatureEntry.playerName}
          armorClass={creatureEntry.playerArmorClass}
          width="28"
        />
        <ButtonContainer>
          <EditButtonStyled onClick={handleEditClick}>Edit</EditButtonStyled>
          <DeleteButtonStyled
            onClick={handleDeleteClick}
            id={creatureEntry._id}
          >
            Delete
          </DeleteButtonStyled>
        </ButtonContainer>
      </PlayerEntryStyled>
    </>
  )
  function handleDeleteClick() {
    handleDelete(creatureEntry._id)
  }
  function handleEditClick() {
    handleEdit(creatureEntry._id)
  }
}

const PlayerEntryStyled = styled.div`
  margin: 20px;
  padding: 20px;
  display: grid;
  grid-gap: 5px 20px;
  justify-content: space-between;
  grid-template-columns: 1fr 2fr 2fr;
  background: hsla(208, 40%, 28%, 1);
  background: linear-gradient(
      180deg,
      hsla(208, 40%, 28%, 1) 0%,
      hsla(208, 40%, 23%, 1) 51%,
      hsla(207, 38%, 18%, 1) 100%
    )
    fixed no-repeat;
  color: white;
  border-radius: 5px;
  box-shadow: 4px 4px 6px rgba(8, 15, 21, 0.3);

  h2 {
    font-size: 1.5rem;
  }
`
const ButtonContainer = styled.div`
  grid-column: span 3;
  justify-self: center;
`

const DeleteButtonStyled = styled.button`
  color: var(--dark-red);
  background: var(--light-red);
  border-style: none;
  border-radius: 5px;
  padding: 5px 15px;
  margin: 5px 0 0 5px;
  width: fit-content;
`
const EditButtonStyled = styled.button`
  color: var(--dark-green);
  background: var(--light-green);
  border-style: none;
  border-radius: 5px;
  padding: 5px 20px;
  margin: 5px 5px 0 0;
  width: fit-content;
`
