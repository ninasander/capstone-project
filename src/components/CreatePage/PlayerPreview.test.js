import React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import PlayerPreview from '../CreatePage/PlayerPreview'
import 'jest-styled-components'

describe.only('PlayerPreview', () => {
  const creatureEntry = {
    playerName: 'Player 1',
    playerArmorClass: 17,
    playerInitiative: 20,
    _id: 10,
  }

  it('displays the player name', () => {
    const { getByText } = render(
      <PlayerPreview
        handleDelete={jest.fn()}
        handleEdit={jest.fn()}
        creatureEntry={creatureEntry}
        playerName={creatureEntry.playerName}
        playerArmorClass={creatureEntry.playerArmorClass}
        playerInitiative={creatureEntry.playerInitiative}
        key={creatureEntry._id}
      />
    )

    expect(getByText(creatureEntry.playerName)).toBeInTheDocument()
  })

  it('displays the player AC', () => {
    const acEntry = 'AC: ' + creatureEntry.playerArmorClass
    const { getByText } = render(
      <PlayerPreview
        handleDelete={jest.fn()}
        handleEdit={jest.fn()}
        creatureEntry={creatureEntry}
        playerName={creatureEntry.playerName}
        playerArmorClass={creatureEntry.playerArmorClass}
        playerInitiative={creatureEntry.playerInitiative}
        key={creatureEntry._id}
      />
    )

    expect(getByText(acEntry)).toBeInTheDocument()
  })

  it('renders correctly', () => {
    const tree = renderer.create(
      <PlayerPreview
        handleDelete={jest.fn()}
        handleEdit={jest.fn()}
        creatureEntry={creatureEntry}
        playerName={creatureEntry.playerName}
        playerArmorClass={creatureEntry.playerArmorClass}
        playerInitiative={creatureEntry.playerInitiative}
        key={creatureEntry._id}
      />
    )
    expect(tree).toMatchSnapshot()
  })
})
