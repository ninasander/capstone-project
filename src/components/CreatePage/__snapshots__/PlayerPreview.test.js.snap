// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerPreview renders correctly 1`] = `
.c0 {
  color: red;
  margin: 0;
  padding: 0;
  background: none;
  border-style: none;
  font-size: 110%;
  position: relative;
  left: 285px;
  top: 35px;
  z-index: 100;
}

.c2 {
  color: green;
  background: lightgreen;
  border-style: none;
  grid-column: span 3;
  border-radius: 5px;
  padding: 5px 20px;
  margin: 0;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  justify-self: center;
}

.c1 {
  font-size: 80%;
  margin: 0 20px;
  padding: 20px;
  display: grid;
  grid-template-columns: 1fr 3fr;
  background: linear-gradient( 180deg,hsla(208,40%,28%,1) 0%,hsla(208,40%,23%,1) 51%,hsla(207,38%,18%,1) 100% ) fixed no-repeat;
  color: white;
  border-radius: 5px;
  box-shadow: 4px 4px 6px rgba(8,15,21,0.3);
}

.c1 h1 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin: 10px 0;
}

.c1 h2 {
  margin: 0;
  font-size: 140%;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c1 p {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin: 0 20px;
}

<div>
  <button
    className="c0"
    id={10}
    onClick={[Function]}
  >
    âœ•
  </button>
  <div
    className="c1"
  >
    <h1>
      20
    </h1>
    <h2>
      Player 1
    </h2>
    <p>
      AC: 
      17
    </p>
    <button
      className="c2"
      onClick={[Function]}
    >
      Edit
    </button>
  </div>
</div>
`;
